MapboxVisionTests/VisionManagerTests.swift

```diff
@@ -0,0 +1,137 @@
+@testable import MapboxVision
+import XCTest
+
+class VisionManagerTests: XCTestCase {
+    var visionManager: VisionManager?
+    var dependencies: VisionDependencies?
+
+    override func setUp() {
+        super.setUp()
+
+        dependencies = VisionDependencies(
+            native: MockNative(),
+            synchronizer: MockSynchronizable(),
+            recorder: MockSessionRecorder(),
+            dataProvider: MockDataProvider(),
+            deviceInfo: DeviceInfoProvider()
+        )
+
+        self.visionManager = VisionManager(dependencies: dependencies!, videoSource: MockVideoSource())
+    }
+
+    func testNoCrashOnVisionManagerDeallocWithoutDestroy() {
+        // We called destroy in destructors of VisionManager, BaseVisionManager, VisionManagerNative and VisionManagerBaseNative,
+        // so VisionManagerBaseNative's destroy method called multiple times, so we tried to destroy already destroyed resources,
+        // which caused an EXC_BAD_ACCESS.
+
```

Comment:
nit: I would remove that line, but it's not an issue.

Reply:
done